File: go/mst/prim_test.go
package mst

import (
	"testing"
)

func TestPrimRandomCases(t *testing.T) {
	for range 10_000 {
		td := newRandomTestData()
		got := PrimMinimumSpanningForest(td.n, append([][]int{}, td.edges...))
		td.assertAnswer(t, "PrimMinimumSpanningForest", got)
	}
}

File: java/src/mst/PrimTest.java
package mst;

import org.junit.Test;

public class PrimTest {
	@Test
	public void testRandomCases() {
		for (int test = 0; test < 10_000; test++) {
			var td = new randomTestData();
			var got = Prim.minimumSpanningForest(td.n, td.edges.clone());
			td.assertAnswer(got);
		}
	}
}
